.gitlab/ci/.gitlab-ci.on-push.yml

```diff
@@ -482,14 +499,15 @@ slack-notify-nightly-build:
     JIRA_PROJECT_ID: $JIRA_PROJECT_ID
     JIRA_ISSUE_UNRESOLVED_TRANSITION_NAME: $JIRA_ISSUE_UNRESOLVED_TRANSITION_NAME
 
-slack-notify-on-push-build:
-  when: always
+slack-notify-on-push:
+  rules:
+    - if: '$$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/ && $CI_COMMIT_BRANCH != $master_branch_name'
```

Comment:
Why not always?

Reply:
push to master has it's own separate slack notify step